decisions/0001-file-sharing.md

```diff
@@ -0,0 +1,103 @@
+---
+# These are optional metadata elements. Feel free to remove any of them.
+status: "proposed"
+date: 2025-02-24
+decision-makers: tbd
+consulted: tbd
+informed: tbd
+---
+
+# Raw data / file sharing
+
+## Context and Problem Statement
+
+Loculus is currently only able to share consensus sequences but not the raw reads. However, such data are of high value. In case of clinical samples where consensus sequences are available, raw data are useful for deeper analysis, validation of results, and the detection of low-frequency variants. In the case of environmental samples, such as those from wastewater, consensus sequences cannot be derived and raw data are the only way to analyze the sample.
+
+The raw read data of a sample is significantly larger than a consensus sequence. For example, a consensus sequence of a SARS-CoV-2 sample is about 30 KB, while the raw data is around hundred MB to a few GB. We therefore need an approach to store and share large files.
+
+From the user perspective, depending on the instance, it should be possible to upload metadata, consensus sequences and/or raw data. The consensus sequences are, as now, stored in a FASTA. The raw data can be a FASTQ, a BAM file or something else – what exactly one has/can upload depends also on the instance. The maintainer-configurable pipeline will take the user data and return processed data: this could be only consensus sequences and metadata (i.e., some tabular data), or include some other unaligned raw data (e.g., the raw reads after dehumanizing), aligned raw reads, or other type of files. The user will be able to download all processed data. Aligned reads may be passed into SILO for fast querying.
+
+
+## Decision Drivers
+
+* Scalability of storage size: The system should be able to scale to TBs or even PBs of data files.
+* Scalability of traffic: The system should be able to handle the upload and download of large files.
+* Same data privacy, integrity, and persistence model as before: This means that the data uploaded by a group should remain private until it has been approved. Once approved, it should be guaranteed that the originally uploaded data remain available and versioned (unless explicit interventions by a maintainer).
+
```

Comment:
Could you add a section why we decided against also allowing users to drag and drop fies of raw reads? This would be the nicest user experience

Reply:
I don't think we have decided against this?

Reply:
Yeah - _client_ can also be the web frontend, it can also handle the whole back-and-forth of calling endpoints and do it all for you, and you as a _user_ just provide the file.

Reply:
Agreed, we should (at a later stage) certainly also implement the upload of raw data in our website.

Reply:
Drag and drop of folders isn't supported AFAIKT, but we totally would if we could. Maybe in the future in the website we can somehow make single file upload easier with drag an drop or something.